### PR TITLE
[NMS] Add support to disable dhcp service at gateway level

### LINKED ISSUE #2765: Add unmanaged eNB setup option on NMS

### PR DESCRIPTION
**Pull request by karthiksubraveti**
## Summary

This change adds support to disable dhcp service at the gateway
## Test Plan
yarn test passes fine
unmanaged_enb_gif

[skipJenkins] 
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: nms/app/packages/magmalte/app/state/lte/EquipmentState.js [hunk 1]
```diff
@@ -308,6 +310,16 @@ export async function UpdateGateway(props: UpdateGatewayProps) {
       }),
     );
   }
+
+  if (props.dnsConfig !== undefined) {
```
Comment (Scott8440) [inline nms/app/packages/magmalte/app/state/lte/EquipmentState.js]:
```suggestion
  if (props.dnsConfig) {
```

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfig.js [hunk 2]
```diff
@@ -339,14 +339,23 @@ function GatewayRAN({gwInfo}: {gwInfo: lte_gateway}) {
...
       (enbs: {[string]: EnodebInfo}, serial: string) => {
-        if (enbCtx.state.enbInfo[serial] != null) {
+        if (enbCtx?.state?.enbInfo?.[serial] != null) {
           enbs[serial] = enbCtx.state.enbInfo[serial];
         }
         return enbs;
       },
       {},
     ) || {};
+  const dhcpServiceSts = gwInfo.cellular.dns?.dhcp_server_enabled ?? true;
```
Comment (Scott8440) [inline nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfig.js]: I'm just checking that the default state should actually be true. Seems weird that it would be so if this flag didn't exist.

**Review by Scott8440 — APPROVED** [review]

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfig.js (hunk 2 above)
Comment (karthiksubraveti) [reply nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfig.js]: Yes. It does seem a little weird. @ardzoht do you know why we don't have dns configuration by default and set the dhcp_service_enabled as true or instead call the flag as dhcp_service_disabled?

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfig.js [hunk 3]
```diff
@@ -339,14 +339,23 @@ function GatewayRAN({gwInfo}: {gwInfo: lte_gateway}) {
...
   const ran: DataRows[] = [
+    [
+      {
+        category: 'eNodeB Dhcp Service',
```
Comment (andreilee) [inline nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfig.js]: Probably should capitalize all of `DHCP`

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfig.js (hunk 2 above)
Comment (andreilee) [inline nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfig.js]: `dhcpServiceStatus` instead please

File: nms/app/packages/magmalte/app/state/lte/EquipmentState.js [hunk 4]
```diff
@@ -256,22 +257,23 @@ export type UpdateGatewayProps = {
...
+  dnsConfig?: gateway_dns_configs,
   enbs?: enodeb_serials,
   networkId: network_id,
   setLteGateways: ({[string]: lte_gateway}) => void,
 };
 
 export async function UpdateGateway(props: UpdateGatewayProps) {
   const {networkId, gatewayId, setLteGateways} = props;
-  if (props.tierId !== undefined) {
+  if (props.tierId) {
```
Comment (andreilee) [inline nms/app/packages/magmalte/app/state/lte/EquipmentState.js]: Refactoring changes like this should go in separate PRs, ideally.

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfigEdit.js [hunk 5]
```diff
@@ -753,30 +770,45 @@ export function RanEdit(props: Props) {
...
           <AltFormField label={'Transmit Enabled'}>
             <Switch
+              disabled={!(dnsConfig?.dhcp_server_enabled ?? true)}
               onChange={() =>
                 handleRanChange('transmit_enabled', !ranConfig.transmit_enabled)
               }
               checked={ranConfig.transmit_enabled}
             />
           </AltFormField>
+          <AltFormField label={'eNodeB Dhcp Service'}>
```
Comment (andreilee) [inline nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfigEdit.js]: `eNodeB DHCP Service`
